docs/blog/posts/2025-09-16-github-actions-token-exfiltration.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Token Exfiltration Campaign via GitHub Actions Workflows
+description: Incident report of a recent attack campaign targeting GitHub Actions workflows to exfiltrate PyPI tokens, our response, and steps to protect your projects.
+authors:
+  - miketheman
+date: 2025-09-16
+tags:
+  - security
+  - transparency
+---
+
+## Summary
+
+We recently responded to an attack campaign where malicious actors injected code into GitHub Actions workflows 
+attempting to steal PyPI publishing tokens.
+PyPI was not compromised, and no PyPI packages were published by the attackers.
+
+Attackers targeted a wide variety of repositories, many of which had PyPI tokens stored as GitHub secrets,
+modifying their workflows to send those tokens to external servers.
+While the attackers successfully exfiltrated some tokens, they do not appear to have used them on PyPI.
+
+We've invalidated all affected tokens and notified the impacted project maintainers.
+If you're one of them, we have emailed you from <security@pypi.org>.
+
+<!-- more -->
+
+You can read more about the details of the attack on [GitGuardian's blog](https://blog.gitguardian.com/ghostaction-campaign-3-325-secrets-stolen/).
+
+## Timeline and Response
+
+On September 5th, a GitGuardian employee used the [PyPI "Report as malware" button](./2024-03-06-malware-reporting-evolved.md)
+to submit their findings for a project named `fastuuid` - namely they found a malicious GitHub Actions workflow 
+attempting to exfiltrate PyPI tokens to a remote server.
+No compromise on PyPI was found, tokens relating to the user accounts were invalidated,
+and we reached out to the project owners to notify and help secure the account and project.
+
+Later on September 5th, another researcher from GitGuardian emailed PyPI Security 
+directly about their current findings, effectively an expansion of the previous attack.
+Due to some of the contents in that email, it ended up in our inbound Spam folder,
```

Comment:
Seems like this is likely to happen again given what these emails look like, do we have any plan to avoid this in the future?

Reply:
Not right now - the spam filter is something managed outside our control, and most of our inbound reports come via API.
For these folks, I recommended to them that in the future they send a second email without links, but most security teams use `hxxp://` style notation, which they didn't use here.